modules/osdk-building-ansible-operator.adoc

```diff
@@ -0,0 +1,377 @@
+** Set the subject namespace to `REPLACE_NAMESPACE`. This must be changed to the
+namespace in which the Operator is deployed.
+--
+
+. *Customize the Operator logic.*
++
+For this example, the `memcached-operator` executes the following reconciliation
+logic for each `Memcached` Custom Resource (CR):
++
+--
+* Create a `memcached` Deployment if it does not exist.
+* Ensure that the Deployment size is the same as specified by the `Memcached` CR.
+--
++
+By default, the `memcached-operator` watches `Memcached` resource events as
+shown in the `watches.yaml` file and executes the Ansible role `Memcached`:
++
+[source,yaml]
+----
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+----
++
+You can optionally customize the following logic in the `watches.yaml` file:
+
+.. Specifying a `role` option configures the Operator to use this specified path
+when launching `ansible-runner` with an Ansible role. By default, the new
+command fills in an absolute path to where your role should go:
++
+[source,yaml]
+----
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+  role: /opt/ansible/roles/memcached
+----
+
+.. Specifying a `playbook` option in the `watches.yaml` file configures the
+Operator to use this specified path when launching `ansible-runner` with an
+Ansible playbook:
++
+[source,yaml]
+----
+- version: v1alpha1
+  group: cache.example.com
+  kind: Memcached
+  playbook: /opt/ansible/playbook.yaml
+----
+
+. *Build the Memcached Ansible role.*
++
+Modify the generated Ansible role under the `roles/memcached/` directory. This
+Ansible role controls the logic that is executed when a resource is modified.
+
+.. *Define the `Memcached` spec.*
++
+Defining the spec for an Ansible-based Operator can be done entirely in Ansible.
+The Ansible Operator passes all key-value pairs listed in the CR spec field
+along to Ansible as
+link:https://docs.ansible.com/ansible/2.5/user_guide/playbooks_variables.html#passing-variables-on-the-command-line[variables].
+The names of all variables in the spec field are converted to snake case by the
+Operator before running ansible. For example, `serviceAccount` in the spec
+becomes `service_account` in Ansible.
++
+[TIP]
+====
+You should perform some type validation in Ansible on the variables to ensure
+that your application is receiving expected input.
+====
++
+In case the user does not set the `spec` field, set a default by modifying the
+`roles/memcached/defaults/main.yml` file:
++
+[source,yaml]
+----
+size: 1
+----
+
+.. *Define the `Memcached` Deployment.*
++
+With the `Memcached` spec now defined, you can define what Ansible is actually
+executed on resource changes. Because this is an Ansible role, the default
+behavior executes the tasks in the `roles/memcached/tasks/main.yml` file.
++
+The goal is for Ansible to create a Deployment if it does not exist, which runs
+the `memcached:1.4.36-alpine` image. Ansible 2.5+ supports the
```

Comment:
2.5 supports the deprecated `k8s_raw` and `openshift_raw` modules, `k8s` was added in 2.7